File: mediumSpringWebflux/src/main/java/com/example/mediumspringwebflux/board/service/FindAllBoard.java
package com.example.mediumspringwebflux.board.service;

import com.example.mediumspringwebflux.board.dto.BoardResponse;
import com.example.mediumspringwebflux.board.repository.BoardRepository;
import com.example.mediumspringwebflux.board.service.facade.BoardToBoardResponse;
import org.springframework.stereotype.Service;
import reactor.core.publisher.Flux;
import reactor.core.scheduler.Schedulers;

@Service
public class FindAllBoard {
    private final BoardRepository boardRepository;
    private final BoardToBoardResponse boardToBoardResponse;

    public FindAllBoard(BoardRepository boardRepository, BoardToBoardResponse boardToBoardResponse) {
        this.boardRepository = boardRepository;
        this.boardToBoardResponse = boardToBoardResponse;
    }

    public Flux<BoardResponse> execute() {
        return boardRepository.findAll()
                .flatMap(boardToBoardResponse::rapping)
                .subscribeOn(Schedulers.boundedElastic());
    }

}

File: mediumSpringWebfluxcorutine/src/main/kotlin/com/example/mediumspringwebfluxcorutine/board/service/FindAllBoard.kt
package com.example.mediumspringwebfluxcorutine.board.service

import com.example.mediumspringwebfluxcorutine.board.document.Board
import com.example.mediumspringwebfluxcorutine.board.dto.BoardResponse
import com.example.mediumspringwebfluxcorutine.board.repository.BoardRepository
import kotlinx.coroutines.flow.toList
import kotlinx.coroutines.reactive.asFlow
import org.springframework.stereotype.Service

@Service
class FindAllBoard(
    private val boardRepository: BoardRepository
) {

    suspend fun execute(): List<BoardResponse> {
        val boards: List<Board> = boardRepository.findAll()
            .asFlow().toList()

        return boards.map { b -> wrap(b) }
    }

    private fun wrap(board: Board): BoardResponse {
        return BoardResponse(
            id = board.id.toString(),
            title = board.title,
            content = board.content
        )
    }

}